source/core/Date.cpp

```diff
@@ -180,27 +233,35 @@ namespace Vireo {
         time_t rawtime = timestamp.Integer() - kStdDT1970re1904;
         struct tm timeinfo;
         localtime_r(&rawtime, &timeinfo);
-        snprintf(timeZoneAbbr, sizeof(timeZoneAbbr), "%s", timeinfo.tm_zone);
+       if (strchr(timeinfo.tm_zone, ' ') == NULL) {
+          // if timezone name is abbreviated, there won't be space. True on Linux and Mac (native, node.js, and browser)
+          snprintf(timeZoneAbbr, sizeof(timeZoneAbbr), "%s", timeinfo.tm_zone);
+       }
+       else {
+          // if timezone name is unabbreviated (True on Windows (both node.js and browser), then abbreviate it
+          abbreviateTimeZone(timeinfo.tm_zone, timeZoneAbbr);
+       }
 // #elif kVireoOS_emscripten variant deleted (03/2017); localtime_r works correctly
 #elif kVireoOS_windows
-        // Issue #218 TODO - FIXME: return time zone info/dst based on timestamp.Integer(), not current time.
-        // Possibly use SystemTimeToTzSpecificLocalTime
         TIME_ZONE_INFORMATION timeZoneInfo;
-        int rc = GetTimeZoneInformation(&timeZoneInfo);
-        wchar_t *timeZoneName = L"TimeZoneUnknown";
-        if (rc == TIME_ZONE_ID_STANDARD)
+        int rc = GetTimeZoneInformationForYear(year, NULL, &timeZoneInfo);
+        char timeZoneName[kTempCStringLength] = "TODO-TimeZone";
```

Comment:
"TODO" doesn't seem like a string that should be exposed to the user if an error.